src/main/java/appeng/items/tools/powered/ToolWirelessTerminal.java

```diff
@@ -84,26 +155,162 @@ public void addCheckedInformation( final ItemStack stack, final EntityPlayer pla
 		}
-		else
+
+		return false;
+	}
+
+	private void updateIsUsable( final ItemStack stack, final Entity entity )
+	{
+		if( stack.hasTagCompound() )
+		{
+			final NBTTagCompound tag = Platform.openNbtData( stack );
+			if( tag != null )
+			{
+				boolean usable = ( this.hasPower( null, 0.5, stack ) && ToolWirelessTerminal.isLinked( stack ) );
+				if( usable )
+				{
+					final long parsedKey = Long.parseLong( this.getEncryptionKey( stack ) );
+					final ILocatable securityStation = AEApi.instance().registries().locatable().getLocatableBy( parsedKey );
+					if( securityStation == null )
+					{
+						usable = false;
+					}
+				}
+				if( usable )
+				{
+					usable = rangeCheck( stack, entity );
+				}
+				tag.setBoolean( "WirelessUsable", usable );
+				tag.setInteger( "untilUpdate", 20 );
+			}
+		}
+	}
+
+	/**
+	 * Checks if the user can connect to the ME system associated with the Wireless Terminal
+	 *
+	 * @param itm The wireless terminal
+	 * @param player the player trying to connect
+	 *
+	 * @return
+	 */
+	public boolean rangeCheck( ItemStack itm, final Entity player )
+	{
+		//Get the encryption key
+		ILocatable obj = null;
+		try
+		{
+			final long encKey = Long.parseLong( this.getEncryptionKey( itm ) );
+			obj = AEApi.instance().registries().locatable().getLocatableBy( encKey );
+		}
+		catch( final NumberFormatException err )
+		{
+			// :P
+		}
+
+		IGrid grid = null;
+		IMEMonitor<IAEItemStack> inv = null;
+		if( obj instanceof IGridHost )
+		{
+			final IGridNode n = ( (IGridHost) obj ).getGridNode( ForgeDirection.UNKNOWN );
+			if( n != null )
+			{
+				grid = n.getGrid();
+				if( grid != null )
+				{
+					IStorageGrid sg = grid.getCache( IStorageGrid.class );
+					if( sg != null )
+					{
+						inv = sg.getItemInventory();
+					}
+				}
+			}
+		}
+
+		if( grid != null && inv != null )
+		{
+
+			final IMachineSet tw = grid.getMachines( TileWireless.class );
+
+			for( final IGridNode n : tw )
+			{
+				final IWirelessAccessPoint wap = (IWirelessAccessPoint) n.getMachine();
+				if( this.testWap( wap, player ) )
+				{
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Returns if you can connect to a WAP
+	 *
+	 * @param wap The WAP to test if you can connect to
+	 * @param player The player entity
+	 *
+	 * @return True if the WAP is within range and active
+	 */
+	private boolean testWap( final IWirelessAccessPoint wap, final Entity player )
+	{
+		double rangeLimit = wap.getRange();
+		rangeLimit *= rangeLimit;
+
+		final DimensionalCoord dc = wap.getLocation();
+
+		if( dc.getWorld().provider.dimensionId == player.worldObj.provider.dimensionId )
 		{
-			lines.add( StatCollector.translateToLocal( "AppEng.GuiITooltip.Unlinked" ) );
+			final double offX = dc.x - player.posX;
+			final double offY = dc.y - player.posY;
+			final double offZ = dc.z - player.posZ;
+
+			final double r = offX * offX + offY * offY + offZ * offZ;
+			if( r < rangeLimit )
+			{
+				if( wap.isActive() )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)